sw/device/lib/dif/dif_spi_device.c

```diff
@@ -77,219 +77,51 @@ dif_spi_device_result_t dif_spi_device_configure(
   tx_fifo_bounds = bitfield_field32_write(
       tx_fifo_bounds, SPI_DEVICE_TXF_ADDR_LIMIT_FIELD, tx_fifo_end);
 
-  spi->rx_fifo_len = config.rx_fifo_len;
-  spi->tx_fifo_len = config.tx_fifo_len;
```

Comment:
One thing to note here is that, unlike every other IP's DIFs, the SPI Device DIFs had been written to embed the `rx_fifo_len` and `tx_fifo_len` into the DIF context (handle) struct: `dif_spi_device_t`, see below in `dif_spi_device.h` file on lines 116--117. These are configuration parameters that are also embedded in the DIF configuration struct: `dif_spi_device_config_t`, and just copied over to `dif_spi_device_t` struct here to cache them, even though they can be deduced from values held in CSRs after the SPI Device has been configured. 

In order to auto-generate the IRQ DIFs, and others in the future, the `dif_spi_device_t` struct must be auto-generated, as described in #8142. Therefore, this struct must be similar across all IPs, meaning these values must not be stored in the `dif_spi_device_t` struct. Rather, if they are used throughout any of the SPI Device's DIFs, they must passed in as arguments (contained in a different struct/context) or read from the CSRs within a DIF. To stay consistent with a similar design for IP's with `*_params_t` struct, e.g., the Alert Handler (see #8409), and suggestions made by @moidx in regards to isolating stateful DIF contexts, the former design choice was taken. However, this can be changed as described. Or another approach can be taken in anyone has a different alternative in mind.

This means that some DIFs below will now take in an additional `const dif_spi_device_config_t *config` argument.